origin-discovery/src/apollo/resolvers.js

```diff
@@ -57,28 +58,40 @@ const resolvers = {
     },
 
     async offers (root, args, context, info) {
-      const opts = {}
-      opts.buyerAddress = args.buyerAddress
-      opts.listingId = args.listingId
-      const offers = search.Offer.search(opts)
+      const clause = {}
+      if (args.listingId) {
+        clause.listingId = args.listingId
+      }
+      if (args.buyerAddress) {
+        clause.buyerAddress = args.buyerAddress.toLowerCase()
+      }
+      if (args.sellerAddress) {
+        clause.sellerAddress = args.sellerAddress.toLowerCase()
+      }
+      const rows = await db.Offer.findAll({ where: clause })
+      const offers = rows.map(offer => offer.data)
       return { nodes: offers }
     },
 
     async offer (root, args, context, info) {
-      return search.Offer.get(args.id)
+      const row = await db.Offer.findByPk({ id: args.id })
+      return (row.length === 0) ? row.data : null
     },
 
     user (root, args, context, info) {
-      return search.User.get(args.walletAddress)
+      // FIXME(franck): some users did not get indexed in prod due to a bug in attestations.
+      // For now only return the address until data gets re-indexed.
+      return { walletAddress: args.walletAddress }
```

Comment:
Could we return null if user is not found and a user object when it is?

Reply:
The problem is that search.User.get(args.walletAddress) throws an exception if it does not find the address.... I could have a try/catch statement but since currently the DApp is not query user data from graphQL I was thinking ok to only return the wallet address. And I plan on fixing the issue in the listener causing some of those user to not get indexed (I'm actually thinking putting the data in a new "user" table in the DB rather than elastic since I don't think we'll nee full text search on users ?

Reply:
Postgres has [some full text search capabilities](https://www.postgresql.org/docs/9.5/textsearch.html) in case we need them. We could copy `firstName` and `lastName` to a `name` full text enabled field. And querying should work there. Even if user has multiple first/last names and we query only by part (one of the multiple names) of the first/last name.

Reply:
Agreed, makes sense. Let's not worry too much about it now - looks like we have plenty of good option if needed in future... :)